cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/BatchSQLEngineAdapter.java

```diff
@@ -671,4 +700,39 @@ protected static void countStageMetrics(StageMetrics stageMetrics,
       return sqlEngine.transform(sqlContext);
     }));
   }
+
+  /**
+   * Try to write the output directly to the SQLEngineOutput registered by this engine.
+   * @param datasetName dataset to write
+   * @param sqlEngineOutput output instance created by this engine
+   * @return boolean specifying if the output was written or not.
+   */
+  public boolean write(String datasetName, SQLEngineOutput sqlEngineOutput) {
```

Comment:
Make it a job as well to follow join/push/pull

Reply:
I've updated the implementation to generate a SQLEngineJob for write operations.